code/modules/clothing/chameleon/chameleon_action_subtypes.dm

```diff
@@ -0,0 +1,281 @@
+	return ..()
+
+/datum/action/item_action/chameleon/change/id
+	chameleon_type = /obj/item/card/id/advanced
+	chameleon_name = "ID Card"
+
+/datum/action/item_action/chameleon/change/id/New(Target)
+	. = ..()
+	if(!istype(target, /obj/item/card/id/advanced/chameleon))
+		stack_trace("Adding chameleon ID action to non-chameleon id ([target])")
+		qdel(src)
+
+/datum/action/item_action/chameleon/change/id/update_item(obj/item/picked_item)
+	. = ..()
+	var/obj/item/card/id/advanced/chameleon/agent_card = target
+	var/obj/item/card/id/copied_card = picked_item
+
+	// If the outfit comes with a special trim override, we'll steal some stuff from that.
+	var/new_trim = initial(copied_card.trim)
+
+	if(new_trim)
+		SSid_access.apply_trim_to_chameleon_card(agent_card, new_trim, TRUE)
+
+	// If the ID card hasn't been forged, we'll check if there has been an assignment set already by any new trim.
+	// If there has not, we set the assignment to the copied card's default as well as copying over the the
+	// default registered name from the copied card.
+	if(!agent_card.forged)
+		if(!agent_card.assignment)
+			agent_card.assignment = initial(copied_card.assignment)
+
+		agent_card.registered_name = initial(copied_card.registered_name)
+
+	agent_card.icon_state = initial(copied_card.icon_state)
+	if(ispath(copied_card, /obj/item/card/id/advanced))
+		var/obj/item/card/id/advanced/copied_advanced_card = copied_card
+		agent_card.assigned_icon_state = initial(copied_advanced_card.assigned_icon_state)
+
+	agent_card.update_label()
+	agent_card.update_appearance(UPDATE_ICON)
+
+/datum/action/item_action/chameleon/change/id/apply_job_data(datum/job/job_datum)
+	var/obj/item/card/id/advanced/chameleon/agent_card = target
+	agent_card.forged = TRUE
+
+	// job_outfit is going to be a path.
+	var/datum/outfit/job/job_outfit = job_datum.outfit
+	if(!job_outfit)
+		return
+
+	// copied_card is also going to be a path.
+	var/obj/item/card/id/copied_card = initial(job_outfit.id)
+	if(!copied_card)
+		return
+
+	// If the outfit comes with a special trim override, we'll use that. Otherwise, use the card's default trim. Failing that, no trim at all.
+	var/new_trim = initial(job_outfit.id_trim) ? initial(job_outfit.id_trim) : initial(copied_card.trim)
+
+	if(new_trim)
+		SSid_access.apply_trim_to_chameleon_card(agent_card, new_trim, FALSE)
+	else
+		agent_card.assignment = job_datum.title
+
+	agent_card.icon_state = initial(copied_card.icon_state)
+	if(ispath(copied_card, /obj/item/card/id/advanced))
+		var/obj/item/card/id/advanced/copied_advanced_card = copied_card
+		agent_card.assigned_icon_state = initial(copied_advanced_card.assigned_icon_state)
+
+	agent_card.update_label()
+	agent_card.update_appearance(UPDATE_ICON)
+
+// /datum/action/item_action/chameleon/change/id_trim
+
+// /datum/action/item_action/chameleon/change/id_trim/New(Target)
```

Comment:
Need to figure out what this is because it was unused when I got here but seems important